mongo/client_bulk_write_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) MongoDB, Inc. 2024-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo
+
+import (
+	"testing"
+
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/internal/assert"
+	"go.mongodb.org/mongo-driver/v2/internal/require"
+)
+
+func TestBatches(t *testing.T) {
+	t.Parallel()
+
+	t.Run("test Addvancing", func(t *testing.T) {
+		t.Parallel()
+
+		batches := &modelBatches{
+			models: make([]clientWriteModel, 2),
+		}
+		batches.AdvanceBatches(3)
+		size := batches.Size()
+		assert.Equal(t, 0, size, "expected: %d, got: %d", 1, size)
+	})
+	t.Run("test appendBatches", func(t *testing.T) {
+		t.Parallel()
+
+		client, err := newClient()
+		require.NoError(t, err, "NewClient error: %v", err)
+		batches := &modelBatches{
+			client: client,
+			models: []clientWriteModel{
+				{"ns0", nil},
+				{"ns1", &ClientInsertOneModel{
+					Document: bson.D{{"foo", 42}},
+				}},
+				{"ns2", &ClientReplaceOneModel{
+					Filter:      bson.D{{"foo", "bar"}},
+					Replacement: bson.D{{"foo", "baz"}},
+				}},
+				{"ns1", &ClientDeleteOneModel{
+					Filter: bson.D{{"qux", "quux"}},
+				}},
+			},
+			offset: 1,
+			result: &ClientBulkWriteResult{
+				Acknowledged: true,
+			},
+		}
+		var n int
+		n, _, err = batches.AppendBatchSequence(nil, 4, 16_000, 16_000)
```

Comment:
What is the significance of `16_000` here?

Reply:
It is just a number big enough not to cut the document, so only the `maxCount` regulates the output. Will add a comment there.